sdk/storage/Azure.Storage.DataMovement/src/LocalTransferCheckpointer.cs

```diff
@@ -397,5 +387,49 @@ public override Task<List<string>> GetStoredTransfersAsync(CancellationToken can
                 throw Errors.MissingTransferIdCheckpointer(transferId);
             }
         }
+
+        /// <summary>
+        /// Takes the path of the checkpointer reads all the files in the top directory level
+        /// and populates the _transferStates
+        /// </summary>
+        private void InitializeExistingCheckpointer()
+        {
+            // Retrieve all valid checkpointer files stored in the checkpointer path.
+            foreach (string path in Directory.EnumerateFiles(_pathToCheckpointer, "*", SearchOption.TopDirectoryOnly)
+                .Where(f => Path.HasExtension(string.Concat(
+                    DataMovementConstants.PlanFile.FileExtension,
+                    DataMovementConstants.PlanFile.SchemaVersion))))
+            {
+                // Ensure each file has the correct format
+                if (JobPartPlanFileName.TryParseJobPartPlanFileName(path, out JobPartPlanFileName partPlanFileName))
```

Comment:
Will `TryParseJobPartPlanFileName` throw if encounters a parsing error? If so, maybe that is fine, if not, do we want to do something in the `else` here?

I guess that leads to the question what do we actually want to do if we encounter a "broken" file? Do we want to crash or do we want to just skip that transfer and log a warning? Or something else?

Reply:
I think the correct thing to do is to just skip over it. If it only has the correct extension, but the file name format isn't correctly we should just skip it.

Though I think you're right, we should at least log a warning? Like "we detected this file with the correct extension but the file name was not in the format we expected." We can make a TODO for that cause we need to pull in a ClientDiagnostics here to do the logging, and it's a little difficult.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/36285